incubator/hnc/config/webhook/webhook_patch.yaml

```diff
@@ -7,6 +7,11 @@ webhooks:
 - name: objects.hnc.x-k8s.io
   timeoutSeconds: 2
   sideEffects: None
+  namespaceSelector:
```

Comment:
Should we put this on the namespace webhook too? If not, why not? Maybe comment why this only applies to certain namespaces here.

Reply:
Ah right, no I should put it on namespace validator too. Initially, I added the `namespaceSelector` to all validators and realized for HNC CRDs, e.g. `hc` or `anchor`, we should not filter excluded namespaces since we want to deny setting excluded namespaces in the hierarchy. So I removed the `namespaceSelector` on them, but I should keep it on the namespace validator (just tested it that if not, updates to the e.g. `kube-system` namespace will also get denied when webhook service is down and filtering it out won't affect blocking the excluded namespaces to be part of the hierarchy).

Will add it back on the namespace validtor.

Reply:
Ah the issue is that if I have it filtered, it won't block user adding `hnc.x-k8s.io/excluded-namespace=true` to other non-excluded namespaces. Let me look into how to work it around.

Reply:
Discussed offline with @adrianludwin, here's an example if having `hnc.x-k8s.io/excluded-namespace` label value `NotIn` `true` on the namespace validator:
```
$ k label ns default hnc.x-k8s.io/excluded-namespace=false
Error from server (Forbidden): admission webhook "namespaces.hnc.x-k8s.io" denied the request: The namespace contains an illegal label "hnc.x-k8s.io/excluded-namespace", which should only be applied on the excluded namespaces set in the deployment configuration

$ k label ns default hnc.x-k8s.io/excluded-namespace=true
namespace/default labeled
```
, that mismatching label is denied by the webhook but the matching label is exempt from validation.

I will keep the `namespaceSelector` for namespace validator and remove the label validation. On the other hand, remove the excluded-namespace label from the non-excluded namespaces in our reconcilers.

Reply:
sgtm. That's really unfortunate behaviour IMO in K8s but it is what it is!

Reply:
Added comment for only adding `namespaceSelector` on object validator but not on namespace validator:
```
  # We only filter out excluded namespaces from this object validator so that
  # when HNC (webhook service specifically) is down, operations in the excluded
  # namespaces won't be affected. Validators on HNC CRs are not filtered because
  # they are supposed to prevent abuse of HNC CRs in excluded namespaces.
  # Namespace validator is not filtered to prevent abuse of excluded namespace.
  # When HNC is down, updates on excluded namespace are forbidden, but those who
  # can update namespaces like `kube-system` should likely be able to delete the
  # VWHConfiguration to make the updates
```